src/jet/mod.rs

```diff
@@ -137,6 +137,7 @@ pub trait Jet: Copy + Eq + Ord + Hash + std::fmt::Debug + std::fmt::Display {
                 c_writeBit(&mut a_frame, bit);
             }
         }
+        mac.back(src_ty_bit_width);
```

Comment:
0e0e13f287052b18d4e8e09e94124f338781b7f3: Is that the semantics of jets? They peek their input and don't advance the read cursor? This doesn't seem to be documented in the tech report (the jet chapter is empty). All I can see is that the test from the next commit fails on master, and that somehow the other tests are unaffected, which is surprising. We probably used `1 → A` jets almost all of the time.

Reply:
Literally nothing advances the read cursor in Simplicity and leaves it advanced.